models/inventory/PurchaseReceipt.ts

```diff
@@ -1,11 +1,17 @@
 import { ListViewSettings } from 'fyo/model/types';
 import { getTransactionStatusColumn } from 'models/helpers';
+import { updateSerialNoStatus } from './helpers';
 import { PurchaseReceiptItem } from './PurchaseReceiptItem';
 import { StockTransfer } from './StockTransfer';
 
 export class PurchaseReceipt extends StockTransfer {
   items?: PurchaseReceiptItem[];
 
+  async afterSubmit(): Promise<void> {
```

Comment:
Why is this here, the same logic is duplicated across all `StockTransfer` subclasses. This should be handled there.